src/providers/solana/cachedRpcFactory.ts

```diff
@@ -0,0 +1,126 @@
+import { CacheType } from "../utils";
+import { jsonReplacerWithBigInts, jsonReviverWithBigInts } from "../../utils";
+
+export class CachedSolanaRpcFactory extends SolanaClusterRpcFactory {
+  public readonly getTransactionCachePrefix: string;
+
+  // Holds the underlying transport that the cached transport wraps.
+  protected rateLimitedTransport: RpcTransport;
+
+  // RPC client based on the rate limited transport, used internally to check confirmation status.
+  protected rateLimitedRpcClient: RpcFromTransport<SolanaRpcApiFromTransport<RpcTransport>, RpcTransport>;
+
+  constructor(
+    providerCacheNamespace: string,
+    readonly redisClient?: CachingMechanismInterface,
+    ...rateLimitedConstructorParams: ConstructorParameters<typeof RateLimitedSolanaRpcFactory>
+  ) {
+    // SolanaClusterRpcFactory shares the last two constructor parameters with RateLimitedSolanaRpcFactory.
+    const superParams = rateLimitedConstructorParams.slice(-2) as [
+      ConstructorParameters<typeof SolanaClusterRpcFactory>[0], // clusterUrl: ClusterUrl
+      ConstructorParameters<typeof SolanaClusterRpcFactory>[1], // chainId: number
+    ];
+    super(...superParams);
+
+    // Create the rate limited transport and RPC client.
+    const rateLimitedRpcFactory = new RateLimitedSolanaRpcFactory(...rateLimitedConstructorParams);
+    this.rateLimitedTransport = rateLimitedRpcFactory.createTransport();
+    this.rateLimitedRpcClient = rateLimitedRpcFactory.createRpcClient();
+
+    // Pre-compute as much of the redis key as possible.
+    const cachePrefix = `${providerCacheNamespace},${new URL(this.clusterUrl).hostname},${this.chainId}`;
+    this.getTransactionCachePrefix = `${cachePrefix}:getTransaction,`;
+  }
+
+  public createTransport(): RpcTransport {
+    return async <TResponse>(...args: Parameters<RpcTransport>): Promise<TResponse> => {
+      const { method, params } = args[0].payload as { method: string; params?: unknown[] };
+      const cacheType = this.redisClient ? this.cacheType(method) : CacheType.NONE;
+
+      if (cacheType !== CacheType.NONE) {
+        const redisKey = this.buildRedisKey(method, params);
+
+        // Attempt to pull the result from the cache.
+        const redisResult = await this.redisClient?.get<string>(redisKey);
+
+        // If cache has the result, parse the json and return it.
+        if (redisResult) {
+          return JSON.parse(redisResult, jsonReviverWithBigInts);
+        }
+
+        // Cache does not have the result. Query it directly and cache it if finalized.
+        return this.requestAndCacheFinalized<TResponse>(...args);
+      }
+
+      return this.rateLimitedTransport<TResponse>(...args);
+    };
+  }
+
+  private async requestAndCacheFinalized<TResponse>(...args: Parameters<RpcTransport>): Promise<TResponse> {
+    const { method, params } = args[0].payload as { method: string; params?: unknown[] };
+
+    // Only handles getTransaction right now.
+    if (method === "getTransaction") {
```

Comment:
Nit: You could use an early return `if (method !== "getTransaction") return this.rateLimitedTransport<TResponse>(...args);`
So we don't need the else at the end and improve readability